community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -63,53 +63,26 @@
 
 
 def instance_properties(nodeset, model, placement_group, labels=None):
-    template = lkp.node_template(model)
-    template_info = lkp.template_info(template)
-
     props = NSDict()
 
-    slurm_metadata = {
-        "slurm_cluster_name": cfg.slurm_cluster_name,
-        "slurm_instance_role": "compute",
-        "startup-script": (
-            Path(cfg.slurm_scripts_dir or util.dirs.scripts) / "startup.sh"
-        ).read_text(),
-        "VmDnsSetting": "GlobalOnly",
-    }
-    info_metadata = {
-        item.get("key"): item.get("value") for item in template_info.metadata["items"]
-    }
+    if labels: # override labels on instance and disks
+        template = lkp.node_template(model)
+        template_info = lkp.template_info(template)
 
-    props_metadata = {**info_metadata, **slurm_metadata}
-    props.metadata = {
-        "items": [NSDict({"key": k, "value": v}) for k, v in props_metadata.items()]
-    }
-
-    labels = {
-        "slurm_cluster_name": cfg.slurm_cluster_name,
-        "slurm_instance_role": "compute",
-        **(labels or {}),
-    }
-    props.labels = {**template_info.labels, **labels}
-
-    for disk in template_info.disks:
-        # do not label local ssd
-        if (
-            "diskType" not in disk.initializeParams
-            or disk.initializeParams.diskType == "local-ssd"
-        ):
-            continue
-        disk.initializeParams.labels.update(labels)
-    props.disks = template_info.disks
+        props.labels = {**template_info.labels, **labels}
+        
+        for disk in template_info.disks:
```

Comment:
> I think for disk in may need to be de-dented one level

I don't see why. We update `template_info.disks` IFF extra labels are supplied. After that we set `props.disks = template_info.disks` 
![Screenshot 2024-07-30 at 6 31 37 PM](https://github.com/user-attachments/assets/f082891e-baa6-4092-9140-4cfd155b4244)

> It's not clear to me why we are using labels here vs props.labels (I realize this was true before). The important thing is that all disks are labeled with the same minimum set that enables cost-tracking.

Because we want labels on the disk specifically, AFAIK disk will not inherit labels from the instance. I will double check the inheritance.

Reply:
UPD: Performed manual testing - additional disks DO NOT inherit labels from the instance. We need to set them explicitly.

Reply:
Offline discussion we have confirmed that the set of labels is not changing: ghpc_deployment always + job_id when exclusive is set.